src/lib/RpcApi.ts

```diff
@@ -150,11 +155,24 @@ export default class RpcApi {
                 let account;
                 // Simply testing if the property exists (with `'walletId' in request`) is not enough,
                 // as `undefined` also counts as existing.
-                if (request && (request as ParsedSimpleRequest).walletId) {
-                    account = await WalletStore.Instance.get((request as ParsedSimpleRequest).walletId);
-                    if (!account) {
-                        // no account found although it's required
-                        this.reject(new Error('Account ID not found'));
+                if (request) {
+                    let accountRequired;
+                    if ((request as ParsedSimpleRequest).walletId) {
+                        accountRequired = true;
+                        account = await WalletStore.Instance.get((request as ParsedSimpleRequest).walletId);
+                    } else if (requestType === RequestType.SIGN_TRANSACTION) {
+                        accountRequired = true;
+                        const address = (request as ParsedSignTransactionRequest).sender;
+                        account = this._store.getters.findWalletByAddress(address.toUserFriendlyAddress());
+                    } else if (requestType === RequestType.SIGN_MESSAGE) {
+                        accountRequired = false; // Sign message allows user to select an account
+                        const address = (request as ParsedSignMessageRequest).signer;
+                        if (address) {
+                            account = this._store.getters.findWalletByAddress(address.toUserFriendlyAddress());
+                        }
+                    }
```

Comment:
The `CHECKOUT` flow is missing here, similar to sign_message, where a `sender` is not required but can be included (added with #214, after you started this branch).

Additionally, checkout has a `forceSender` option, which should throw an error when the passed `sender` does not exist. Thus I would say to set `accountRequired = request.forceSender` to trigger the correct error behavior.